synapse/storage/data_stores/main/room.py

```diff
@@ -448,21 +448,32 @@ def _get_media_mxcs_in_room_txn(self, txn, room_id):
         """
         mxc_re = re.compile("^mxc://([^/]+)/([^/#?]+)")
 
-        next_token = self.get_current_events_token() + 1
+        next_token = None
         local_media_mxcs = []
         remote_media_mxcs = []
 
-        while next_token:
-            sql = """
-                SELECT stream_ordering, json FROM events
-                JOIN event_json USING (room_id, event_id)
-                WHERE room_id = ?
-                    AND stream_ordering < ?
-                    AND contains_url = ? AND outlier = ?
-                ORDER BY stream_ordering DESC
-                LIMIT ?
-            """
-            txn.execute(sql, (room_id, next_token, True, False, 100))
+        while True:
+            if next_token is None:
+                sql = """
+                    SELECT stream_ordering, json FROM events
+                    JOIN event_json USING (room_id, event_id)
+                    WHERE room_id = ?
+                        AND contains_url = ? AND outlier = ?
+                    ORDER BY stream_ordering DESC
+                    LIMIT ?
+                """
+                txn.execute(sql, (room_id, True, False, 100))
+            else:
+                sql = """
+                    SELECT stream_ordering, json FROM events
+                    JOIN event_json USING (room_id, event_id)
+                    WHERE room_id = ?
+                        AND stream_ordering < ?
+                        AND contains_url = ? AND outlier = ?
+                    ORDER BY stream_ordering DESC
+                    LIMIT ?
```

Comment:
can we reduce the redundancy between the sql statements? Maybe have a `%(where_clauses)s` which we can interpolate or something?